crates/dotrain/src/parser/rainlangdocument/logic.rs

```diff
@@ -685,6 +685,17 @@ impl RainlangDocument {
                 lhs_alias: None,
                 id: None,
             }))?;
+        } else if NUMERIC_PATTERN.is_match(next) {
+            if HEX_PATTERN.is_match(next) && next.len() % 2 == 1 {
+                self.problems
+                    .push(ErrorCode::OddLenHex.to_problem(vec![], next_pos));
```

Comment:
<img width="695" alt="Screenshot 2024-05-17 at 20 56 23" src="https://github.com/rainlanguage/dotrain/assets/629710/02f5a176-8d2c-4810-a803-13c4b617934f">

already exists onchain, don't need it here